docarray/typing/tensor/abstract_tensor.py

```diff
@@ -213,3 +214,18 @@ def _docarray_to_json_compatible(self):
         :return: a representation of the tensor compatible with orjson
         """
         ...
+
+    @abc.abstractmethod
+    def reshape(self, shape: Tuple[int, ...]):
+        """
+        Gives a new shape to tensor without changing its data.
+        :return: a tensor with the same data and number of elements as self
+            but with the specified shape.
+        """
+        ...
+
+    @property
+    @abc.abstractmethod
+    def shape(self) -> Tuple[int, ...]:
+        """The shape of this tensor."""
+        ...
```

Comment:
we should move these to the computation backend IMO

Reply:
okei